tests/functional_tests/util_resources.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 The Monero Project
+# 
+# All rights reserved.
+# 
+# Redistribution and use in source and binary forms, with or without modification, are
+# permitted provided that the following conditions are met:
+# 
+# 1. Redistributions of source code must retain the above copyright notice, this list of
+#    conditions and the following disclaimer.
+# 
+# 2. Redistributions in binary form must reproduce the above copyright notice, this list
+#    of conditions and the following disclaimer in the documentation and/or other
+#    materials provided with the distribution.
+# 
+# 3. Neither the name of the copyright holder nor the names of its contributors may be
+#    used to endorse or promote products derived from this software without specific
+#    prior written permission.
+# 
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+# THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+    Help determine how much CPU power is available at the given time
+    by running numerical calculations
+"""
+
+from __future__ import print_function
+import datetime
+import multiprocessing
+import psutil
+import math
+import decimal
+
+def available_ram_gb():
+    ram_bytes = psutil.virtual_memory().available
+    kilo = 1024
+    ram_gb = ram_bytes / kilo / kilo / kilo
+    return ram_gb
+
+def pi_digits_lib(digits):
+    """
+    Returns the number of digits of Pi
+    """
+    return float('{:.{}f}'.format(math.pi, digits))
+
+def pi_decimal(digits):
+    """
+        Calculate PI using Leibnitz' formula
+        
+        https://en.wikipedia.org/wiki/Leibniz_formula_for_%CF%80
+        https://proofwiki.org/wiki/Leibniz's_Formula_for_Pi
+    """
+
+    pi = 0.0
+    n = 4.0
+    d = 1.0
+    pi_lib = pi_digits_lib(digits)
```

Comment:
Wouldn't it be better to just run a fixed number of iterations? Who even cares whether this converges on pi or not?

Reply:
Yes, I was considering this.

A greater concern to me is however, that I get much faster calculation of the same algo under Python3 locally, than in the Python2, so you will see a huge overshoot in the CI's logs. So if we don't update to Python3, then even this algo doesn't seem right. I'm only worried, that other Python scripts won't work after the update. I'll let it process for the night and get back here.

Reply:
I think you're inevitably going to see fluctuations with your current approach, as, if you look at, e.g. microbenchmark data from different python3 point releases, it's all over the map (some get faster, some slower). You'd be much better off running a benchmark closer to the machine (i.e. written in C). I suggest decimal because it's a thin wrapper around the C mpdec library, but, unfortunately, it could also fall back to pure-python if mpdec isn't installed, and, in any case, it's not as good as some vanilla C program.

Reply:
I can write a C++ app. I prefer not to touch  threads in C

Reply:
Done. The C++ app delivers very good results.

Still to do for the app: Copyright and documentation.